### PR TITLE
`Publish Review` buttons should indicate why they are disabled

### LINKED ISSUE #23547: Buttons in review box should indicate reason for disablement
### Description

When trying to review one's own PR, the UI buttons are disabled and there is no indication on why. I think a simple "You can not approve you own Pull Request" would go a long way in making it clear.

### Screenshots

image

### PR DESCRIPTION
**Pull request by puni9869**
Adding tooltip on "Approve" and "Review Changes" buttons. is try to do by them self on their PR.
Resolves https://github.com/go-gitea/gitea/issues/23547
`Approve` tootip text :- "Pull request authors can’t approve their own pull request"
`Request Changes` tooltip text:- "Pull request authors can’t request changes on their own pull request"

Before 
https://user-images.githubusercontent.com/115237/225984020-306c048f-cb9b-4dee-8929-1c74c331518b.png

After 
image

### CODE REVIEW

**Comment by silverwind** [comment]
Note: Can't backport because of new translations.

File: templates/repo/diff/new_review.tmpl [hunk 1]
```diff
@@ -22,9 +22,21 @@
 					</div>
 				{{end}}
 				<div class="ui divider"></div>
-				<button type="submit" name="type" value="approve" {{if and $.IsSigned ($.Issue.IsPoster $.SignedUser.ID)}} disabled {{end}} class="ui submit green tiny button btn-submit">{{$.locale.Tr "repo.diff.review.approve"}}</button>
+				{{if and $.IsSigned ($.Issue.IsPoster $.SignedUser.ID)}}
+					<span class="ui tooltip" data-content="{{$.locale.Tr "repo.diff.review.self_approve"}}">
+						<button type="submit" name="type" value="approve" disabled class="ui submit green tiny button btn-submit">{{$.locale.Tr "repo.diff.review.approve"}}</button>
+					</span>
```
Comment (silverwind) [inline templates/repo/diff/new_review.tmpl:29]: These `span` wrappings can be removed, just put the `tooltip` class and the `data-content` on the button itself.

File: templates/repo/diff/new_review.tmpl [hunk 2]
```diff
@@ -22,9 +22,21 @@
...
+				{{else}}
+					<button type="submit" name="type" value="approve" class="ui submit green tiny button btn-submit">{{$.locale.Tr "repo.diff.review.approve"}}</button>
+				{{end}}
 				<button type="submit" name="type" value="comment" class="ui submit tiny basic button btn-submit">{{$.locale.Tr "repo.diff.review.comment"}}</button>
-				<button type="submit" name="type" value="reject" {{if and $.IsSigned ($.Issue.IsPoster $.SignedUser.ID)}} disabled {{end}} class="ui submit red tiny button btn-submit">{{$.locale.Tr "repo.diff.review.reject"}}</button>
+				{{if and $.IsSigned ($.Issue.IsPoster $.SignedUser.ID)}}
+					<span class="ui tooltip" data-content="{{$.locale.Tr "repo.diff.review.self_approve"}}">
```
Comment (jolheiser) [inline templates/repo/diff/new_review.tmpl]:
```suggestion
					<span class="ui tooltip" data-content="{{$.locale.Tr "repo.diff.review.self_reject"}}">
```

File: options/locale/locale_en-US.ini [hunk 3]
```diff
@@ -2265,7 +2265,9 @@ diff.review.header = Submit review
 diff.review.placeholder = Review comment
 diff.review.comment = Comment
 diff.review.approve = Approve
+diff.review.self_reject = Pull request authors can’t request changes on their own pull request
 diff.review.reject = Request changes
+diff.review.self_approve = Pull request authors can’t approve their own pull request
```
Comment (jolheiser) [inline options/locale/locale_en-US.ini]:
```suggestion
diff.review.self_reject = Pull request authors can't request changes on their own pull request
diff.review.reject = Request changes
diff.review.self_approve = Pull request authors can't approve their own pull request
```

**Review by jolheiser — CHANGES_REQUESTED** [review]

File: templates/repo/diff/new_review.tmpl [hunk 4]
```diff
@@ -22,9 +22,21 @@
 					</div>
 				{{end}}
 				<div class="ui divider"></div>
-				<button type="submit" name="type" value="approve" {{if and $.IsSigned ($.Issue.IsPoster $.SignedUser.ID)}} disabled {{end}} class="ui submit green tiny button btn-submit">{{$.locale.Tr "repo.diff.review.approve"}}</button>
+				{{if and $.IsSigned ($.Issue.IsPoster $.SignedUser.ID)}}
```
Comment (silverwind) [inline templates/repo/diff/new_review.tmpl]: We can create a in-template variable for this value, avoiding re-computation and repetition.

```handlebars
{{$showSelfTooltip := (and $.IsSigned ($.Issue.IsPoster $.SignedUser.ID))}}
```

And then use as `{{if $showSelfTooltip}}`

File: templates/repo/diff/new_review.tmpl (hunk 1 above)
Comment (puni9869) [reply templates/repo/diff/new_review.tmpl:29]: As it is a issue https://github.com/atomiks/tippyjs/issues/286 and fixed in later release.


I tried to do so  but we have to programatically handle the  in tippyjs
something like this `instance.enable();`

https://atomiks.github.io/tippyjs/v6/methods/#enable

Applying tooltip in disable state.
image

**Comment by puni9869** [comment]
> Note: Can't backport because of new translations.

What should be my next step. Kinda new to process.

File: templates/repo/diff/new_review.tmpl (hunk 1 above)
Comment (silverwind) [reply templates/repo/diff/new_review.tmpl:29]: Ah, indeed, I recall this is likely intentional behaviour of tippy, so lets keep it. When altering DOM structure, we need to ensure no JS depends on the structure, but as this isn't meant to do anything when disabled, we should be save.

**Comment by silverwind** [comment]
> > Note: Can't backport because of new translations.
> 
> What should be my next step. Kinda new to process.

Nothing from your side. Once the PR is merged in `main`, translations will start coming in over time from Crowdin. Once enough translations are there, we can opt to backport into 1.19, but as this is not a critical enhancement, I'd say it's not worth the effort.

File: options/locale/locale_en-US.ini (hunk 3 above)
Comment (puni9869) [reply options/locale/locale_en-US.ini]: Done

File: templates/repo/diff/new_review.tmpl (hunk 2 above)
Comment (puni9869) [reply templates/repo/diff/new_review.tmpl]: Done

File: templates/repo/diff/new_review.tmpl (hunk 4 above)
Comment (puni9869) [reply templates/repo/diff/new_review.tmpl]: Done

File: templates/repo/diff/new_review.tmpl [hunk 5]
```diff
@@ -22,9 +22,22 @@
 					</div>
 				{{end}}
 				<div class="ui divider"></div>
-				<button type="submit" name="type" value="approve" {{if and $.IsSigned ($.Issue.IsPoster $.SignedUser.ID)}} disabled {{end}} class="ui submit green tiny button btn-submit">{{$.locale.Tr "repo.diff.review.approve"}}</button>
+				{{$showSelfTooltip := (and $.IsSigned ($.Issue.IsPoster $.SignedUser.ID))}}
+				{{if $showSelfTooltip}}
+					<span class="ui tooltip" data-content="{{$.locale.Tr "repo.diff.review.self_approve"}}">
```
Comment (silverwind) [inline templates/repo/diff/new_review.tmpl]:
```suggestion
					<span class="gt-dib tooltip" data-content="{{$.locale.Tr "repo.diff.review.self_approve"}}">
```

Fixes the tooltip position:

image

See https://github.com/go-gitea/gitea/blob/main/web_src/css/helpers.css for what it does.

File: templates/repo/diff/new_review.tmpl [hunk 6]
```diff
@@ -22,9 +22,22 @@
...
+						<button type="submit" name="type" value="approve" disabled class="ui submit green tiny button btn-submit">{{$.locale.Tr "repo.diff.review.approve"}}</button>
+					</span>
+				{{else}}
+					<button type="submit" name="type" value="approve" class="ui submit green tiny button btn-submit">{{$.locale.Tr "repo.diff.review.approve"}}</button>
+				{{end}}
 				<button type="submit" name="type" value="comment" class="ui submit tiny basic button btn-submit">{{$.locale.Tr "repo.diff.review.comment"}}</button>
-				<button type="submit" name="type" value="reject" {{if and $.IsSigned ($.Issue.IsPoster $.SignedUser.ID)}} disabled {{end}} class="ui submit red tiny button btn-submit">{{$.locale.Tr "repo.diff.review.reject"}}</button>
+				{{if $showSelfTooltip}}
+					<span class="ui tooltip" data-content="{{$.locale.Tr "repo.diff.review.self_reject"}}">
```
Comment (silverwind) [inline templates/repo/diff/new_review.tmpl]:
```suggestion
					<span class="gt-dib tooltip" data-content="{{$.locale.Tr "repo.diff.review.self_reject"}}">
```

**Review by silverwind — APPROVED** [review]

**Review by jolheiser — APPROVED** [review]

**Review by yardenshoham — APPROVED** [review]

**Review by delvh — APPROVED** [review]
I'd also prefer it if we didn't need the span.
However, as you mentioned that it is apparently a tippy "bug"/intended behavior, so I'll allow it.

**Comment by silverwind** [comment]
> I'd also prefer it if we didn't need the span. However, as you mentioned that it is apparently a tippy "bug"/intended behavior, so I'll allow it.

Yes, it is actually documented as well here: 

https://github.com/atomiks/tippyjs/blob/4fbef2cb4e66ad240cc0a67936c7300fee16f03d/website/src/pages/v6/constructor.mdx#disabled-elements

**Pull request merged**